src/Query.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, {
+  ObservableQuery,
+  ApolloQueryResult,
+  ApolloError,
+  FetchMoreOptions,
+  UpdateQueryOptions,
+  FetchMoreQueryOptions,
+} from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { ZenObservable } from 'zen-observable-ts';
+
+import shallowEqual from './shallowEqual';
+const invariant = require('invariant');
+const pick = require('lodash.pick');
+
+import { QueryOpts, OperationVariables } from './types';
+import { parser, DocumentType } from './parser';
+
+export interface QueryRenderProp {
+  error?: ApolloError;
+  networkStatus: number;
+  loading: boolean;
+  variables: OperationVariables;
+  fetchMore: (
+    fetchMoreOptions: FetchMoreQueryOptions & FetchMoreOptions,
+  ) => Promise<ApolloQueryResult<any>>;
+  refetch: (variables?: OperationVariables) => Promise<ApolloQueryResult<any>>;
+  startPolling: (pollInterval: number) => void;
+  stopPolling: () => void;
+  updateQuery: (
+    mapFn: (previousQueryResult: any, options: UpdateQueryOptions) => any,
+  ) => void;
+}
+
+export interface QueryProps {
+  query: DocumentNode;
+  options?: QueryOpts;
+  skip?: Boolean;
+  loading?: () => React.ReactNode;
+  error?: (error: ApolloError) => React.ReactNode;
+  render?: ((result: QueryRenderProp) => React.ReactNode);
+  children?: ((result: QueryRenderProp) => React.ReactNode) | React.ReactNode;
```

Comment:
This should be function only, no `| React.ReactNode` - it only confuses the possibilities.